lib/bowlingScoreCard.js

```diff
@@ -0,0 +1,62 @@
+class BowlingScoreCard {
+  constructor() {
+    this.eachRollScoreArray = [];
+  }
+
+  roll = (pinDown) => {
+    this.eachRollScoreArray.push(pinDown);
+  };
+
+  overallScore = () => {
+    let score = 0;
+    let rollNumber = 0;
+    for (let i = 0; i < 10; i++) {
+      if (this.isASpareFrame(rollNumber)) {
+        score += 10 + this.spareBonusScore(rollNumber);
+        rollNumber += 2;
+      } else if (this.isAStrikeFrame(rollNumber)) {
+        score += 10 + this.strikeBonusScore(rollNumber);
+        rollNumber += 1;
+      } else {
+        score += this.sumPinsDownInFrame(rollNumber);
+        rollNumber += 2;
+      }
+    }
+    if (this.isASpareFrame(rollNumber - 2)) {
+      score += this.eachRollScoreArray[rollNumber];
+    }
+    return score;
+  };
+
+  sumPinsDownInFrame = (rollNumber) => {
+    return (
+      this.eachRollScoreArray[rollNumber] +
+      this.eachRollScoreArray[rollNumber + 1]
+    );
+  };
+
+  isASpareFrame = (rollNumber) => {
+    return (
+      this.eachRollScoreArray[rollNumber] +
+        this.eachRollScoreArray[rollNumber + 1] ===
+      10
+    );
```

Comment:
I think it could make the code slightly cleaner and more readable if you built some of these methods into a Frame class.

Reply:
Completely agree with splitting this into a frame class. I had the same thought as I was going along.